src/aspect.ts

```diff
@@ -1,27 +1,80 @@
 import { Handle } from './interfaces';
 import { createHandle } from './lang';
 
+/**
+ * An object that provides the necessary APIs to be MapLike
+ */
+export interface MapLike<K, V> {
+	get(key: K): V;
+	set(key: K, value?: V): this;
+}
+
+/**
+ * An internal type guard that determines if an value is MapLike or not
+ *
+ * @param value The value to guard against
+ */
+function isMapLike(value: any): value is MapLike<any, any> {
+	return value && typeof value.get === 'function' && typeof value.set === 'function';
+}
+
+export interface Indexable {
```

Comment:
I hadn't realised that was there...  The method I was using was keeping it close to the first use site, as I find it makes the code easier to read that way.

Do you feel ordering them alphabetically, when they are also used within the module, improves readability and maintainability, or were you just being pedantic?  If it is the former, then I would like to understand it better.  If it is the latter, it is easy to change the style guide.